api/envoy/extensions/transport_sockets/tls/v3/tls.proto

```diff
@@ -124,8 +125,24 @@ message DownstreamTlsContext {
   OcspStaplePolicy ocsp_staple_policy = 8 [(validate.rules).enum = {defined_only: true}];
 }
 
+// TLS key log configuration.
+// The key log file format is "format used by NSS for its SSLKEYLOGFILE debugging output" (text taken from openssl man page)
+message TlsKeyLog {
+  // The path to save the TLS key log.
+  string tls_keylog_path = 1 [(validate.rules).string = {min_len: 1}];
+
+  // The src IP address that will be used to filter the connection which should save the TLS key log
+  // If it is not set, any src IP address will be matched.
+  config.core.v3.Address tls_keylog_src = 2;
+
+  // The destination IP address that will be used to filter the connection which should save the TLS key log
+  // If it is not set, any dst IP address will be matched.
+  config.core.v3.Address tls_keylog_dst = 3;
+  // If none of src and dst is set, no key log will be saved for any connections.
```

Comment:
Maybe it would make more sense to say that if neither one is specified, we save key logs for *all* connections?  That way, src and dst basically just act as filters, which may be a bit more intuitive.

Alternatively, I'm fine with leaving this as-is, but I think this comment should be moved up above the src and dst fields, and it should say something like "at least one of src or dst must be specified", and we should reject the config as invalid if neither field is present.

Reply:
fixed